addons/hr_expense/models/product_product.py

```diff
@@ -24,3 +24,14 @@ def _compute_standard_price_update_warning(self):
                             "There are unposted expenses linked to this category. Updating the category cost will change expense amounts. "
                             "Make sure it is what you want to do."
                         )
+
+    def write(self, vals):
+        result = super().write(vals)
+        if 'standard_price' in vals:
+            expenses_sudo = self.env['hr.expense'].sudo().search([('product_id', 'in', self.ids), ('state', 'in', ['reported', 'draft'])])
+            for expense_sudo in expenses_sudo:
+                expense_sudo.write({
+                    'product_has_cost': expense_sudo.product_id and not expense_sudo.company_currency_id.is_zero(expense_sudo.product_id.standard_price),
+                    'product_has_tax': bool(expense_sudo.product_id.supplier_taxes_id.filtered_domain(self.env['account.tax']._check_company_domain(expense_sudo.company_id))),
```

Comment:
Not sure about the _check_company_domain here either. Is there any advantage in using that function instead of writing the domain directly ?

Reply:
Honestly I'm not entirely sure, maybe it's shorter and easier to read this way? I can see that it's always used to filter taxes in the codebase. I guess here it won't really make a difference. I'll change it if you want :)